### PR TITLE
moves join pairs helper methods to own class

### PR DESCRIPTION
**Pull request by seut**

### CODE REVIEW

**Comment by seut** [comment]
@matriv pls review

File: sql/src/main/java/io/crate/analyze/relations/JoinPairs.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import io.crate.planner.node.dql.join.JoinType;
+import io.crate.sql.tree.QualifiedName;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class JoinPairs {
```
Comment (matriv) [inline sql/src/main/java/io/crate/analyze/relations/JoinPairs.java]: Maybe also declare final since it's a util class.

**Comment by matriv** [comment]
Igtm otherwise

**Comment by seut** [comment]
@matriv pls check fixup

**Comment by matriv** [comment]
Igtm

**Pull request merged**